src/components/queryBuilder/utils.ts

```diff
@@ -259,6 +263,195 @@ export const getSQLFromQueryOptions = (options: SqlBuilderOptions): string => {
+      filters.push({ condition: 'AND' } as Filter);
+      continue;
+    } else if (c.trim().toUpperCase() === 'OR') {
+      filters.push({ condition: 'OR' } as Filter);
+      continue;
+    }
+    const stringPhrases = c.match(/([''])(?:(?=(\\?))\2.)*?\1/g)?.map((x) => (x = x.substring(1, x.length - 1)));
+    const phrases = c.match(/(\w+|\$(\w+)|!=|<=|>=|=)/g);
+    if (!phrases) {
+      continue;
+    }
+    const isNotClause = phrases[0] === 'NOT';
+    const opAndVal = getOperatorAndValues(phrases, stringPhrases ? stringPhrases : []);
+    filters[filters.length - 1] = {
+      ...filters[filters.length - 1],
+      filterType: 'custom',
+      key: isNotClause ? phrases[1] : phrases[0],
+      operator: opAndVal.f ? opAndVal.f : '',
+      value: opAndVal.v ? opAndVal.v : '',
+      type: getFilterType(c, stringPhrases),
+    } as Filter;
+  }
+  return filters;
+}
+
+function getOperatorAndValues(phrases: string[], stringPhrases: string[]): { f: FilterOperator; v: any } {
+  if (isWithInTimeRangeFilter(phrases)) {
+    return {
+      f: phrases[0] === 'NOT' ? FilterOperator.OutsideGrafanaTimeRange : FilterOperator.WithInGrafanaTimeRange,
+      v: '',
+    };
+  }
+
+  let op = phrases[1];
+  if (Object.values(FilterOperator).includes(op.toUpperCase() as FilterOperator)) {
+    return {
+      f: op.toUpperCase() as FilterOperator,
+      v: stringPhrases.length > 0 ? stringPhrases : phrases.slice(2, phrases.length),
+    };
+  }
+  for (let i = 2; i < phrases.length; i++) {
+    op += ` ${phrases[i]}`;
+    if (Object.values(FilterOperator).includes(op.toUpperCase() as FilterOperator)) {
+      return {
+        f: op.toUpperCase() as FilterOperator,
+        v: stringPhrases.length > 0 ? stringPhrases : phrases.slice(i + 1, phrases.length),
+      };
+    }
+  }
+  return { f: '' as FilterOperator, v: null };
+}
+
+function getFilterType(whereClause: string, stringPhrases?: string[]): '' | 'datetime' | 'date' | 'string' {
+  if (stringPhrases && stringPhrases.length > 0) {
+    return 'string';
+  }
+  if (whereClause.includes(':date:YYYY-MM-DD')) {
+    return 'date';
+  }
+  if (whereClause.includes(':date')) {
+    return 'datetime';
+  }
+  return '';
+}
+
+function isWithInTimeRangeFilter(phrases: string[]): boolean {
+  if (!phrases || phrases.length === 0) {
+    return false;
+  }
+  let hasFrom = false;
+  let hasTo = false;
+  for (const p of phrases) {
+    if (p === '__from') {
+      hasFrom = true;
+    } else if (p === '__to') {
+      hasTo = true;
+    }
+  }
+  return hasFrom && hasTo;
+}
+
+function getMetricsFromAst(selectClauses: Clause[]): { metrics: BuilderMetricField[]; fields: string[] } {
+  const metrics: BuilderMetricField[] = [];
+  const fields: string[] = [];
+  for (const c of selectClauses) {
+    if (!isString(c) || !c.trim() || c.trim() === ',') {
+      continue;
+    }
+    let isMetric = false;
+    for (const agg of Object.values(BuilderMetricFieldAggregation)) {
```

Comment:
probably another case where .filter and .map would make this a bit more readable